File: src/pub_sub/pub_sub/listener_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class Listener_node(Node):
    def __init__(self):
        super().__init__("listener_node")
        self.subscriber_ = self.create_subscription(String, "topic", self.listener_callback, 10)
    
    def listener_callback(self, msg):
        self.get_logger().info(f'Received {msg.data}')


        

def main(args = None):
    rclpy.init(args = args)
    # create
    listner_node = Listener_node()
    # use
    rclpy.spin(listner_node)
    # destroy
    listner_node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: src/talker_listener/src/listener_node.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
using std::placeholders::_1;

class Listener_node : public rclcpp::Node{
  public:
    Listener_node() : Node("My_listener") {
      subscription_ = this->create_subscription<std_msgs::msg::String>("PUB", 10, std::bind(&Listener_node::topic_callback, this, _1));
    }

  private:
    void topic_callback(const std_msgs::msg::String & msg) const {
      RCLCPP_INFO(this->get_logger(), "I heard: '%s'", msg.data.c_str());
    }
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char * argv[]){
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Listener_node>());
  rclcpp::shutdown();
  return 0;
}